src/qr.nim

```diff
@@ -113,38 +103,65 @@ proc qrSvgGenerate(qrcodeData: ptr uint8, svgSize: int32, border: cint, fgColor
     while x < size + border:
 
       if getModule(qrcodeData, x, y):
-        output.add("<use xlink:href='#p' x='" & $svgX & "' y='" & $svgY & "'/>")
-      else:
-        discard
+        output.add("<use xlink:href='#p' x='" & $svgX.int & "' y='" & $svgY.int & "'/>")
 
       inc(x)
       svgY += svgYadd
 
     inc(y)
     svgX += svgYadd
 
-  return svgOpen & output & svgClose
+  # bgColor and fgColor is formatted at compile-time. Sizes and data is formatted at run-time.
+  result = static("""<?xml version="1.0" encoding="utf-8"?><svg width="$$1" height="$$1" viewBox="0 0 $$1 $$1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:ev="http://www.w3.org/2001/xml-events"><rect x="0" y="0" width="$$1" height="$$1" fill="$1"/><defs><rect id="p" width="$$2" height="$$2"/></defs><g fill="$2">$$3</g></svg>""" % [bgColor, fgColor]
```

Comment:
Multiple `$` inserted, e.g. `$$2`

Reply:
Yes, because `bgColor` and `fgColor` is inserted at compile-time, data and sizes is inserted at run-time,
at compile time `$$2` is escaped to `$2`, then `$2` is inserted at run-time, see https://play.nim-lang.org/#ix=2Hi8

Reply:
Thanks for explanation :)